### PR TITLE
Visitor can view the articles on the landing page

### PR DESCRIPTION
**Pull request by Siianko**
### PT story ###
https://www.pivotaltracker.com/story/show/159731281

### User Story ###
```
As an Editor,
In order to catch people's interest,
I would like to have the articles displayed on the landing page.
````

### What have we learned 
How to deal with date in created_at

### CODE REVIEW

**Review by Andycutter — APPROVED** [review]
Looks fine with me

**Review by ESteinman — APPROVED** [review]
LGTM

File: app/views/articles/index.html.haml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+- @articles.each do |article|
+  %p= article.title
+  %p= article.body
+  %p= article.created_at
```
Comment (tochman) [inline app/views/articles/index.html.haml]: Please format the date display

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get 'articles/index'
```
Comment (tochman) [inline config/routes.rb]: Please remove auto generated routes. Also,  follow the standard using `resources` if you need to generate routes for a resource/model.

File: features/step_definitions/background_steps.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Given("the following article exists") do |table|
+    table.hashes.each do |article|
+        Article.create!(article)
```
Comment (tochman) [inline features/step_definitions/background_steps.rb]: Please use a factory

File: features/visitor_can_view_articles_landing_page.feature [hunk 4]
```diff
@@ -0,0 +1,19 @@
+Feature: Visitor can view articles on the landing page
+    As an Editor,
+    In order to catch people's interest,
+    I would like to have the articles displayed on the landing page.
+
+    Background:
+        Given the following article exists
+        | title          | body                                                             | author          | created_at                    |
+        | This is so sad | A recent report suggest that news are mostly sad. Which is sad.  | Pablo Guera     | 0018-08-16 00:00:00 UTC       |
```
Comment (tochman) [inline features/visitor_can_view_articles_landing_page.feature]: Please do not set the date in a table. If you want to be in control of the server date, please consider using a gem like [TimeCop](https://github.com/travisjeffery/timecop) or [DeLoarean](https://github.com/bebanjo/delorean)

**Review by tochman — CHANGES_REQUESTED** [review]

File: app/models/article.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+class Article < ApplicationRecord
+  validates_presence_of :title, :body, :author
```
Comment (tochman) [inline app/models/article.rb]: We had a discussion about author in one of the review sessions. Why is it present on this iteration? There is no mention about the author in this story on PT

File: app/views/articles/index.html.haml [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+  %p= article.author
```
Comment (tochman) [inline app/views/articles/index.html.haml]: ???

File: db/migrate/20180816081644_create_articles.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class CreateArticles < ActiveRecord::Migration[5.2]
+  def change
+    create_table :articles do |t|
+      t.string :title
+      t.text :body
+      t.string :author
```
Comment (tochman) [inline db/migrate/20180816081644_create_articles.rb]: Again, see the PT story (and consequent stories) about the `author` issue See https://www.pivotaltracker.com/story/show/159732606

File: features/visitor_can_view_articles_landing_page.feature (hunk 4 above)
Comment (tochman) [inline features/visitor_can_view_articles_landing_page.feature]: Please remove author from this table

File: spec/models/article_spec.rb [hunk 8]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe Article, type: :model do
+  describe 'DB table' do
+    it { is_expected.to have_db_column :id }
+    it { is_expected.to have_db_column :title }
+    it { is_expected.to have_db_column :body }
+    it { is_expected.to have_db_column :author }
```
Comment (tochman) [inline spec/models/article_spec.rb]: --

File: features/step_definitions/background_steps.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+Given("the following article exists") do |table|
+    table.hashes.each do |article_hash|
+        FactoryBot.create(:article, article_hash)
```
Comment (tochman) [inline features/step_definitions/background_steps.rb]: You can shorten this to omit `FactoryBot`  You are including the Syntax methods in you Cucumber configuration ( `features/support/env.rb/)

File: spec/models/article_spec.rb [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :title }
+    it { is_expected.to validate_presence_of :body } 
+  end
+
+  describe 'Factory' do
+    it 'should have valid Factory' do
+      expect(FactoryBot.create(:article)).to be_valid
```
Comment (tochman) [inline spec/models/article_spec.rb]: Same thing here. You can omit the `FactoryBot` word. Your RailsHelper includes the Syntax::Methods

File: features/step_definitions/background_steps.rb (hunk 9 above)
Comment (Siianko) [reply features/step_definitions/background_steps.rb]: fixed

File: spec/models/article_spec.rb (hunk 10 above)
Comment (Siianko) [reply spec/models/article_spec.rb]: fixed

**Review by tochman — APPROVED** [review]
I'll approve this now. You are still not using Chrome driver. It's not a blocker though. Good job. 
Is till want to see more collaboration on writing code. We will look for that in the future PR's. Also, read each others code carefully.
